src/web/mjs/engine/DiscordPresence.mjs

```diff
@@ -0,0 +1,167 @@
+const discordPresenceId = '726702836775256094';
+const DiscordRPC = require('discord-rpc');
+DiscordRPC.register(discordPresenceId);
+
+export default class DiscordPresence {
+
+    constructor(settings) {
+        this.rpc = null;
+        this.IpcBytes = 0; // keep track of IPC connection
```

Comment:
No need to use a filed when only used in a single method.
Use a scoped variable in the corresponding function.

Reply:
I'm not sure if this is possible without waiting. By keeping track globally I can check the IPC state one interval later which basically allows 15 seconds to pass.

Reply:
OK i understand the intention now